UI/win-dll-blocklist.c

```diff
@@ -0,0 +1,300 @@
+/******************************************************************************
+    Copyright (C) 2023 by Richard Stanway
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+******************************************************************************/
+
+#include <Windows.h>
+#include <psapi.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include "detours.h"
+
+// Undocumented NT structs / function definitions !
+typedef enum _SECTION_INHERIT { ViewShare = 1, ViewUnmap = 2 } SECTION_INHERIT;
+
+typedef enum _SECTION_INFORMATION_CLASS {
+	SectionBasicInformation = 0,
+	SectionImageInformation
+} SECTION_INFORMATION_CLASS;
+
+typedef struct _SECTION_BASIC_INFORMATION {
+	PVOID BaseAddress;
+	ULONG Attributes;
+	LARGE_INTEGER Size;
+} SECTION_BASIC_INFORMATION;
+
+typedef NTSTATUS(STDMETHODCALLTYPE *fn_NtMapViewOfSection)(
+	HANDLE, HANDLE, PVOID, ULONG_PTR, SIZE_T, PLARGE_INTEGER, PSIZE_T,
+	SECTION_INHERIT, ULONG, ULONG);
+
+typedef NTSTATUS(STDMETHODCALLTYPE *fn_NtUnmapViewOfSection)(HANDLE, PVOID);
+
+typedef NTSTATUS(STDMETHODCALLTYPE *fn_NtQuerySection)(
+	HANDLE, SECTION_INFORMATION_CLASS, PVOID, SIZE_T, PSIZE_T);
+
+static fn_NtMapViewOfSection ntMap;
+static fn_NtUnmapViewOfSection ntUnmap;
+static fn_NtQuerySection ntQuery;
+
+#define STATUS_UNSUCCESSFUL ((NTSTATUS)0xC0000001L)
+
+// Method of matching timestamp of DLL in PE header
+typedef enum {
+	TS_IGNORE = 0,      // Ignore timestamp; block all DLLs with this name
+	TS_EQUAL,           // Block only DLL with this exact timestamp
+	TS_LESS_THAN,       // Block all DLLs with an earlier timestamp
+	TS_GREATER_THAN,    // Block all DLLs with a later timestamp
+	TS_ALLOW_ONLY_THIS, // Invert behavior: only allow this specific timestamp
+} ts_compare_t;
+
+typedef struct {
+	// DLL name, lower case
+	const wchar_t *name;
+
+	// Length of name, calculated at startup - leave as zero
+	size_t name_len;
+
+	// PE timestamp
+	const uint32_t timestamp;
+
+	// How to treat the timestamp field
+	const ts_compare_t method;
+} blocked_module_t;
+
+/*
+* Note: The name matches at the end of the string based on its length, this allows
+* for matching DLLs that may have generic names but a problematic version only
+* exists in a certain directory. A name should always include a path component
+* so that e.g. fraps.dll doesn't match notfraps.dll.
+*/
+
+blocked_module_t blocked_modules[] = {
```

Comment:
Can we add `static` qualifier?
```suggestion
static blocked_module_t blocked_modules[] = {
```